espnet2/enh/separator/dc_crn_separator.py

```diff
@@ -84,11 +86,13 @@ def __init__(
         super().__init__()
 
         self._num_spk = num_spk
+        self.predict_noise = predict_noise
         self.mode = mode
         if mode not in ("mapping", "masking"):
             raise ValueError("mode=%s is not supported" % mode)
         self.ref_channel = ref_channel
 
+        num_outputs = self.num_spk + 1 if self.predict_noise else self.num_spk
```

Comment:
Given that all separators has the new attributes `num_outputs ` and `predict_noise `, maybe we should consider adding it to the super class `AbsSeparator`.

Reply:
Not all models can easily support `predict_noise`, such as DPCL and DANet. So I prefer to add support for this function only when it needs minor modifications. Of course, we would recommend newly implemented models to try to support this function.